### ISSUE TITLE
Allow block bindings to be used in patterns, template parts and templates.

### ISSUE THREAD

**Issue by ajgagnon** [WordPress/gutenberg#67287]
## What problem does this address?
Currently, it is possible to use block bindings on posts and custom post type edit pages. The attributes selector does not show up when editing patterns, templates or template parts.

## What is your proposed solution?
Allow block bindings to show up on template parts. Many such block bindings do not have a post as a requirement.

This minimal example:

```php
register_block_bindings_source('my-custom-prefix/current-year', [
    'label' => __('Current year', 'my-custom-prefix'),
    'get_value_callback' => 'my_custom_prefix_current_year',
]);


function my_custom_prefix_current_year()
{
    return date('Y');
}

```

Returning the current year for a footer template part, for example, is a great use case.

**Comment by SantosGuillamot** [comment]
At this moment, the attributes panel only supports the `core/post-meta` source. And in that case, if the post ID and the post type don't exist, there are no fields to connect to. However, if it does, the attributes panel is available. For example, it is available in templates like CPT templates, where we can infer the post type.

For other sources, the idea is to explore how to allow them to reuse it for 6.8, but that isn't ready yet. Is that what you were facing?

**Comment by ajgagnon** [comment]
Yes, that's what I am facing. A custom block bindings source. It also makes sense to have these for templates, right? So in FSE for example, you can output a value based on the metadata of the post. Or perhaps I'm misunderstanding the use-case.

**Comment by SantosGuillamot** [comment]
Yes, that makes sense. Let me try to clarify how it works and what is in the roadmap to ensure we are aligned:

### How it works
- Custom sources don't have a UI yet (unless they implement it), and they have to be added through the Code Editor.
- Post meta source is the only source with that UI. It is shown in places where we have either the `postId` or `postType`. In other cases, there are no custom fields. This means it is available in the post editor (because a `postId` exists) and in FSE templates where we can infer the `postType` (in a custom post type template, for example).

### What is in the roadmap
- Explore how custom sources should hook into this UI.
- Each source will decide where it shows. So if a source doesn't rely on `postId` or `postType` as the example you shared, it would be shown everywhere.
